src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1022,14 +1022,17 @@ module T = struct
       =
     (List.map ~f:(With_status.map ~f:Transaction.forget) a, b, c, d)
 
-  let check_commands ledger ~verifier (cs : User_command.t list) =
+  let check_commands ledger ~verifier (cs : User_command.t With_status.t list) =
     let open Deferred.Or_error.Let_syntax in
     let%bind cs =
       Or_error.try_with (fun () ->
-          List.map cs ~f:(fun cmd ->
+          List.map cs ~f:(fun { With_status.data = cmd; status } ->
               let open Ledger in
-              User_command.to_verifiable ~ledger ~get ~location_of_account cmd
-              |> Or_error.ok_exn ) )
+              let cmd =
+                User_command.to_verifiable ~ledger ~get ~location_of_account cmd
+                |> Or_error.ok_exn
+              in
+              { With_status.data = cmd; status } ) )
```

Comment:
This won't be enough in the case of failed transactions where vk hashes in account updates don't match with what's in the ledger or in the same transaction, right? `Zkapp_command.Verifiable.create` (called by `to_verifiable` here) will fail to find vk hashes, cause a failure here and the block will be rejected

Reply:
Good catch. I've added an optional `allow_missing_vk` argument to `create` that should prevent this from triggering.
Sounds like we probably also need a test; I'll put something together later.